File: examples/zero_config.py
"""Zero-config transform -- GoldenFlow auto-detects and fixes issues.

Usage:
    pip install goldenflow
    python examples/zero_config.py
"""
from __future__ import annotations

import csv
import tempfile
from pathlib import Path

import polars as pl


def create_messy_data() -> Path:
    rows = [
        ["first_name", "last_name", "email", "phone", "city", "amount"],
        ["  John  ", "SMITH", "JOHN@TEST.COM", "(555) 123-4567", "new york", "$1,234.56"],
        ["jane", "  doe  ", "jane@test.com", "5559876543", "  Chicago  ", "999.99"],
        ["BOB", "Wilson", "BOB@TEST.COM", "555.111.2222", "boston", "$50.00"],
        ["Alice", "brown", "alice@test.com", "(555) 333-4444", "AUSTIN", "1,500"],
    ]
    path = Path(tempfile.mktemp(suffix=".csv"))
    with open(path, "w", newline="") as f:
        csv.writer(f).writerows(rows)
    return path


if __name__ == "__main__":
    import goldenflow

    path = create_messy_data()
    print("=" * 60)
    print("GoldenFlow -- Zero-Config Transform")
    print("=" * 60)

    df = pl.read_csv(path)
    print("\nBefore:")
    print(df)

    result = goldenflow.transform_df(df)
    print(f"\nTransformed {len(result.manifest.records)} columns")
    for rec in result.manifest.records:
        print(f"  {rec.column}: {rec.transform} ({rec.affected_rows} rows changed)")

    print("\nAfter:")
    print(result.df)

    path.unlink()

File: examples/typescript/zero-config.ts
/**
 * Zero-config transform — auto-detect and fix messy data.
 *
 * Usage: npx tsx examples/typescript/zero-config.ts
 */

import { TransformEngine } from "goldenflow";

const messyData = [
  {
    name: "  John Smith  ",
    email: "JOHN@EXAMPLE.COM",
    phone: "(555) 123-4567",
    status: "active",
    signup_date: "03/15/2024",
  },
  {
    name: "DR. JANE DOE",
    email: "  jane+work@gmail.com  ",
    phone: "555.987.6543",
    status: "ACTIVE",
    signup_date: "2024-01-20",
  },
  {
    name: "  Bob Johnson  ",
    email: "bob@test.com",
    phone: "+1-555-456-7890",
    status: "actve",
    signup_date: "Jan 5, 2023",
  },
];

const engine = new TransformEngine();
const result = engine.transformDf(messyData);

console.log("=== Zero-Config Transform ===\n");
console.log(`Rows: ${result.rows.length}`);
console.log(`Transforms applied: ${result.manifest.records.length}`);
console.log(`Errors: ${result.manifest.errors.length}\n`);

console.log("--- Transform Audit Trail ---");
for (const record of result.manifest.records) {
  console.log(`  ${record.column}/${record.transform}: ${record.affectedRows}/${record.totalRows} rows`);
}

console.log("\n--- Cleaned Data ---");
for (const row of result.rows) {
  console.log(row);
}
